src/main/java/com/opencart/steps/HeaderPageBL.java

```diff
@@ -104,30 +77,41 @@ public HeaderPageBL verifyMacBookCost(String currencyName) {
         return this;
     }
 
-    /**
-     * Method clicks on Logout button that is located in website's header.
-     *
-     * @return HeaderPageBL
-     */
     public HeaderPageBL clickOnLogoutButton() {
         headerPage.getLogoutButton().click();
         return this;
     }
 
-    public HeaderPageBL clickOnWishListButton() {
+    public WishListPageBL clickOnWishListButton() {
         headerPage.getWishListButton().click();
-        return this;
+        return new WishListPageBL();
     }
 
     public HeaderPageBL clickOnMacbookWishListButton() {
         headerPage.getMacbookWishListButton().click();
         return this;
     }
 
-    public void verifyAddingMacbookToWishList() {
-        String expected = "Wish List (1)";
-        Assert.assertEquals(headerPage.getWishListButton().getText(), expected, "Incorrect value!");
+    public void verifyAddingMacbookToWishListWithoutLogin() {
+        String expected = "You must login or create an account to save MacBook to your wish list!";
+        String actual = headerPage.getSuccessMessage().getText().replaceAll("×", " ").trim();
+        Assert.assertEquals(actual, expected, "You are logged In!");
     }
 
+    public void verifyAddingMacbookInWishListWithLogin() {
+        String expected = "Success: You have added MacBook to your wish list!";
+        String actual = headerPage.getSuccessMessage().getText().replaceAll("×", " ").trim();
+        Assert.assertEquals(actual, expected, "You are not logged in!");
+    }
```

Comment:
That logic also should be moved to MainPageBL, because expected exactly on the Main page